src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HubMessage.cs

```diff
@@ -12,9 +12,8 @@ protected HubMessage()
         {
         }
 
-        // Initialize with capacity 2 for the 2 built in protocols
         private object _lock = new object();
-        private readonly List<SerializedMessage> _serializedMessages = new List<SerializedMessage>(2);
+        private List<SerializedMessage> _serializedMessages;
```

Comment:
Does this save much given we always have at least one protocol enabled? Perhaps building a `SmallList<T>` that uses fields for the first two items would be worth it now?

Reply:
The problem is reads not writes. Allocation up front means we incur this cost even if we never call write.

Reply:
Gotcha